cetlvast/suites/docs/examples/example_06_polymorphic_alloc_deleter.cpp

```diff
@@ -0,0 +1,155 @@
+/// @file
+/// Example of using the cetl::pmr::PolymorphicAllocatorDeleter from cetl/pmr/memory.hpp.
+///
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+///
+/// cSpell: words emplacer
+
+//![example_include]
+#include "cetl/pf17/cetlpf.hpp"
+#include "cetl/pmr/memory.hpp"
+//![example_include]
+
+#include <unordered_map>
+#include <iostream>
+#include <string.h>
+
+class MyObject final
+{
+public:
+    MyObject(const char* name, std::size_t name_length)
+        : name_(nullptr)
+    {
+        name_ = static_cast<char*>(malloc(name_length + 1));
+        strncpy(name_, name, name_length);
```

Comment:
Might want to ensure name_ is null-terminated since strncpy doesn't guarantee it.

Reply:
fixed